src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -249,7 +251,9 @@ class MarathonSchedulerService @Inject() (
 
       //execute tasks, only the leader is allowed to
       migration.migrate()
-      migrationComplete = true
+
+      Await.result(leadershipCoordinator.prepareForStart(), 10.seconds)
```

Comment:
So this waits for all registered WhenLeaderActors? Nice.
Any chance that we do not hard code 10 seconds here?